lib/oban/notifier.ex

```diff
@@ -130,10 +128,19 @@ defmodule Oban.Notifier do
       Oban.Notifier.listen([:gossip, :insert, :signal])
   """
   @spec listen(GenServer.server(), channels :: list(channel())) :: :ok
-  def listen(server \\ __MODULE__, channels) when is_server(server) and is_list(channels) do
+  def listen(server \\ __MODULE__, channels)
+
+  def listen(pid, channels) when is_pid(pid) and is_list(channels) do
     :ok = validate_channels!(channels)
 
-    GenServer.call(server, {:listen, channels})
+    GenServer.call(pid, {:listen, channels})
+  end
+
+  def listen(server, channels) when is_atom(server) and is_list(channels) do
+    Oban.config()
+    |> Map.get(:name, Oban)
+    |> Oban.Registry.whereis(server)
+    |> listen(channels)
```

Comment:
Each Oban instance has its own notifier. This needs to accept an `oban_name` and use the registry from there:

```suggestion
  def listen(oban_name, channels) when is_atom(oban_name) and is_list(channels) do
    oban_name
    |> Registry.whereis(__MODULE__)
    |> listen(channels)
```

Reply:
Okay. I thought you name the notifier and not the Oban instance (just read that you could start more than one). But hat makes sense.